blob/drivertest/drivertest.go

```diff
@@ -1821,6 +1825,75 @@ func testDelete(t *testing.T, newHarness HarnessMaker) {
 	})
 }
 
+// testConcurrentWriteAndRead tests that concurrent writing and reading of
+// keys into a blob works.
+func testConcurrentWriteAndRead(t *testing.T, newHarness HarnessMaker) {
+	ctx := context.Background()
+	h, err := newHarness(ctx, t)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer h.Close()
+	drv, err := h.MakeDriver(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	b := blob.NewBucket(drv)
+	defer b.Close()
+
+	// Prepare data. Each blob dataSize bytes equal to its numeric key.
+	const numKeys = 20
+	const dataSize = 4 * 1024
+	keyData := make(map[int][]byte)
+	for k := 0; k < numKeys; k++ {
+		data := make([]byte, dataSize)
+		for i := 0; i < dataSize; i++ {
+			data[i] = byte(k)
+		}
+		keyData[k] = data
+	}
+
+	blobName := func(k int) string {
+		return fmt.Sprintf("key%d", k)
+	}
+
+	var wg sync.WaitGroup
+
+	// Write all blobs concurrently.
+	for k := 0; k < numKeys; k++ {
+		wg.Add(1)
+		go func(key int) {
+			if err := b.WriteAll(ctx, blobName(key), keyData[key], nil); err != nil {
+				t.Fatal(err)
+			}
+			wg.Done()
+		}(k)
```

Comment:
Add a `defer b.Delete(ctx, blobName(key))` here to clean up the blobs....

Reply:
Done